src/metabase/transform/models/transform.clj

```diff
@@ -14,3 +24,181 @@
+
+;; TODO: This is temporary
+(defn top-schema
+  "Get most used schema"
+  [database-id]
+  ;; Following is temporary. Later do it in nicer way
+  (-> (t2/query {:select [[[:count :id] :c] [:schema :s]]
+                 :from [[:metabase_table :t]]
+                 :where [:= :t.db_id database-id]
+                 :group-by [:schema]
+                 :order-by [[[:count :id] :desc]]
+                 :limit 1})
+      first :s))
+
+;; TODO: This is borrowed. Ideally dataset_query submodule should be created in metabase.queries and this and related
+;;       functions provided there. Gist is that not only functionality metabase.queries is supposed to manipulate this.
+(defn dataset-query->query
+  "Convert the `dataset_query` to pMBQL query."
+  ([dataset-query]
+   (if (-> dataset-query :lib/type #{:mbql/query})
+     dataset-query
+     (some-> (:database dataset-query)
+             lib.metadata.jvm/application-database-metadata-provider
+             (dataset-query->query dataset-query))))
+  ([metadata-provider dataset-query]
+   (if (-> dataset-query :lib/type #{:mbql/query})
+     dataset-query
+     (some->> dataset-query queries/normalize-dataset-query (lib/query metadata-provider)))))
+
+;; TODO: query schema: normalized mbql or native query
+;; query is native query?
+
+(def view-name-prefix
+  "Prefix for view names. Names `mb_transform_\\d+` in user schemas are reserved for managed views."
+  "mb_transform_")
+
+(defn transform-view-name
+  "Generate name for transform view. Id should be unique."
+  [id]
+  (assert (pos-int? id))
+  (str view-name-prefix id))
+
+;; TODO: test for version matching, stub now
+#_{:clj-kondo/ignore [:missing-docstring :unresolved-namespace]}
+(def dataset-query-current-schema @#'metabase.queries.models.card/current-schema-version)
+
+(defn- query-type
+  [query]
+  (or (and (lib/native-stage? query 0)
+           :native)
+      :query))
+
+(defn- assert-unique-name!
+  "Assert there is no same `name` :model/Table in the `schema`."
+  [database schema name]
+  ;; Deliberately not checking :model/TransformView. It is guaranteed the name is unique across that table
+  ;; as the transform_view.id is used to generate name.
+  (when-some [table-ids (not-empty (t2/select-fn-vec :id :model/Table
+                                                     :active true
+                                                     :db_id (:id database)
+                                                     :schema schema
+                                                     :name name))]
+    (throw (ex-info (i18n/tru "View name in use. `{0}<numbers>+` names are reserved." view-name-prefix)
+                    {:status-code 400
+                     :database-id (:id database)
+                     :database-name (:name database)
+                     :schema schema
+                     :name name
+                     :same-name-table-ids table-ids}))))
+
+(defn- assert-unique-display-name!
```

Comment:
as discussed i don't think we should be doing this, but fine to leave in temporarily